src/main/java/io/confluent/idesidecar/restapi/connections/DirectConnectionState.java

```diff
@@ -48,5 +78,126 @@ public Optional<Credentials> getSchemaRegistryCredentials(String clusterId) {
     return Optional.ofNullable(credentials);
   }
 
-  // TODO: DIRECT connections need validation checks
+  @Override
+  public Future<ConnectionStatus> getConnectionStatus() {
+    return Future.join(
+        getKafkaConnectionStatus(),
+        getSchemaRegistryConnectionStatus()
+    ).map(cf -> {
+      var futures = cf.list();
+      var kafkaStatus = (KafkaClusterStatus) futures.get(0);
+      var srStatus = (SchemaRegistryStatus) futures.get(1);
+      return ConnectionStatusBuilder
+          .builder()
+          .kafkaCluster(kafkaStatus)
+          .schemaRegistry(srStatus)
+          .build();
+    });
+  }
+
+  protected Future<KafkaClusterStatus> getKafkaConnectionStatus() {
+    var kafkaConfig = spec.kafkaClusterConfig();
+    if (kafkaConfig == null) {
+      return Future.succeededFuture(
+          ConnectionStatusKafkaClusterStatusBuilder
+              .builder()
+              .state(ConnectedState.NONE)
+              .build()
+      );
+    }
+    // There is a Kafka configuration, so validate the connection by creating an AdminClient
+    // and describing the cluster.
+    try (var adminClient = createAdminClient(kafkaConfig)) {
+      var clusterDesc = adminClient.describeCluster();
+      var actualClusterId = clusterDesc.clusterId().get(5, TimeUnit.SECONDS);
+      // Set the cluster ID
+      getSpec().kafkaClusterConfig().withId(actualClusterId);
+      return Future.succeededFuture(
+          ConnectionStatusKafkaClusterStatusBuilder
+              .builder()
+              .state(ConnectedState.SUCCESS)
+              .build()
+      );
+    } catch (Exception e) {
```

Comment:
I was thinking no, because the message is passed to the status error, and because it is an expected situation if invalid credentials are called. (We don't want users thinking that because the sidecar logged an exception and stack trace that there is a bug within the sidecar causing the failure to connect.)

I could add it a debug log message, though users would never see that and it's not clear it would help us if trying to debug an auth failure.